core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -78,11 +78,11 @@
 
     private final List<ReplicationResponse.ShardInfo.Failure> shardReplicaFailures = Collections.synchronizedList(new ArrayList<>());
 
-    ReplicationOperation(Request request, Primary<Request, ReplicaRequest, PrimaryResultT> primary,
-                         ActionListener<PrimaryResultT> listener,
-                         boolean executeOnReplicas, boolean checkWriteConsistency,
-                         Replicas<ReplicaRequest> replicas,
-                         Supplier<ClusterState> clusterStateSupplier, ESLogger logger, String opType) {
+    public ReplicationOperation(Request request, Primary<Request, ReplicaRequest, PrimaryResultT> primary,
```

Comment:
instead of making it public maybe we move this entire package into `core/src/test/java/org/elasticsearch/index/replication/` or even move everything into `core/src/test/java/org/elasticsearch/index/`